components/dashboards-web-component/src/viewer/components/DashboardReportGenerationCard.jsx

```diff
@@ -138,13 +136,18 @@ export default class DashboardReportGenerationCard extends Component {
 
         html2canvas(document.getElementById('dashboard-container')).then((canvas) => {
             const imgData = canvas.toDataURL('image/png');
-            const w = canvas.width;
-            const h = canvas.height;
-            localStorage.setItem('_dashboard-report:'+currentPage.name,
-                JSON.stringify({ imageData: imgData, width: w, height: h, timestamp:new Date() }));
-            const tempPageList = this.state.pageList.slice();
+            const width = canvas.width;
+            const height = canvas.height;
+            const imageData = JSON.stringify({ imageData: imgData, width: width, height: height, timestamp:new Date() });
+
+            this.capturedPageList.push(imageData);
+            localStorage.setItem('_dashboard-report:'+this.props.dashboardName,JSON.stringify(this.capturedPageList));
+
+            const tempPageList = this.state.pageList;
             tempPageList.push(currentPage.name);
             this.setState({ pageList: tempPageList });
+        }).catch((er) => {
+            console.log(er);
```

Comment:
Instead of a console.log, lets display an error message.
https://v0.material-ui.com/#/components/snackbar